File: 3968-maximum-manhattan-distance-after-all-moves/3968-maximum-manhattan-distance-after-all-moves.java
class Solution {
    public int maxDistance(String moves) {
         int dx = 0, dy = 0, wild = 0;

        for (char ch : moves.toCharArray()) {
            switch (ch) {
                case 'U':
                    dy++;
                    break;
                     case 'D':
                    dy--;
                    break;
                case 'L':
                    dx--;
                    break;
                case 'R':
                    dx++;
                    break;
                case '_':
                    wild++;
                    break;
            }
        }

        return Math.abs(dx) + Math.abs(dy) + wild;
        
    }
}

File: 3968-maximum-manhattan-distance-after-all-moves/3968-maximum-manhattan-distance-after-all-moves.py
class Solution:
    def maxDistance(self, moves: str) -> int:
        x = y = free = 0

        for c in moves:
            if c == 'U':
                y += 1
            elif c == 'D':
                y -= 1
            elif c == 'R':
                x += 1
            elif c == 'L':
                x -= 1
            else:
                free += 1

        return abs(x) + abs(y) + free